packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -1783,6 +1786,47 @@ function updateEffect(
   return updateEffectImpl(PassiveEffect, HookPassive, create, deps);
 }
 
+function mountEvent<T>(callback: () => T): () => T {
+  const hook = mountWorkInProgressHook();
+  const ref = {current: callback};
+
+  function event(...args) {
+    if (isAlreadyRenderingProd()) {
+      throw new Error('An event from useEvent was called during render.');
+    }
+    return ref.current.apply(this, args);
+  }
+
+  mountEffectImpl(
+    UpdateEffect,
+    HookSnapshot,
+    () => {
+      ref.current = callback;
+    },
+    [ref, callback],
+  );
+
+  hook.memoizedState = [ref, event];
```

Comment:
We can do this in a follow up PR but I think we should try to optimize the runtime a bit more. We don't need all the overhead of an effect object since there are no deps and no clean up functions.